### PR TITLE
Add install options pages to docs

### PR DESCRIPTION
**Pull request by davidnuzik**
#### Proposed Changes ####

Add in install options to docs -- this is essentially a copy and paste form existing [k3s install options](https://rancher.com/docs/k3s/latest/en/installation/install-options/). I tweaked where necessary and fixed formatting and links to work with mkdocs. Some content was intentionally removed from the agent/server flags reference pages.

#### Types of Changes ####

Docs

#### Verification ####

When reviewing, please carefully review statements about flags or environment variables to ensure they are indeed cannon. Since I rarely have time for testing these days, I'm a bit out of the loop with rke2 on the command line.

It's also possible I may have missed install env vars that should be called out in docs.

#### Linked Issues ####

Addresses #450 

#### Further Comments ####

After doing this, I'm a fan of mkdocs over hugo (rancher/docs)

### CODE REVIEW

File: docs/install/install_options/how_to_flags/how_to_flags.md [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+Throughout the RKE2 documentation, you will see some options that can be passed in as both command flags and environment variables. The below examples show how these options can be passed in both ways.
+
+### Example: RKE2_KUBECONFIG_MODE
+
+The option to allow writing to the kubeconfig file is useful for allowing an RKE2 cluster to be imported into Rancher. Below are two ways to pass in the option.
+
+Using the flag `--write-kubeconfig-mode 644`:
+
+```bash
+$ curl -sfL https://get.rke2.io | sh -s - --write-kubeconfig-mode 644
```
Comment (cjellick) [inline docs/install/install_options/how_to_flags/how_to_flags.md:14]: dont put dollar signs here. our docs rendered lets you copy these commands out very easily and the dollar sign makes that not work

File: docs/install/install_options/install_options.md [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+The simplest form of this command is as follows:
+```sh
+curl -sfL https://get.rke2.io | sh -
+```
+
+When using this method to install RKE2, the following environment variables can be used to configure the installation:
+
+| Environment Variable | Description |
+|-----------------------------|---------------------------------------------|
+| `INSTALL_RKE2_VERSION` | Version of RKE2 to download from GitHub. Will attempt to download from the latest channel if not specified. |
```
Comment (cjellick) [inline docs/install/install_options/install_options.md:30]:
```suggestion
| <span class='nowrap'>`INSTALL_RKE2_VERSION`</span> | Version of RKE2 to download from GitHub. Will attempt to download from the latest channel if not specified. |
```

File: docs/install/install_options/install_options.md [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+| `INSTALL_RKE2_TYPE` | Type of systemd service to create, can be either "server" or "agent" Default is "server". |
```
Comment (cjellick) [inline docs/install/install_options/install_options.md:31]:
```suggestion
| <span class='nowrap'>`INSTALL_RKE2_TYPE`</span> | Type of systemd service to create, can be either "server" or "agent" Default is "server". |
```

File: docs/install/install_options/install_options.md [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+| `INSTALL_RKE2_CHANNEL_URL` | Channel URL for fetching RKE2 download URL. Defaults to https://update.rke2.io/v1-release/channels. |
```
Comment (cjellick) [inline docs/install/install_options/install_options.md:32]:
```suggestion
| <span class='nowrap'>`INSTALL_RKE2_CHANNEL_URL`</span> | Channel URL for fetching RKE2 download URL. Defaults to https://update.rke2.io/v1-release/channels. |
```

File: docs/install/install_options/install_options.md [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+| `INSTALL_RKE2_CHANNEL` | Channel to use for fetching RKE2 download URL. Defaults to "latest". Options include: `stable`, `latest`, `testing`. |
```
Comment (cjellick) [inline docs/install/install_options/install_options.md:33]:
```suggestion
| <span>`INSTALL_RKE2_CHANNEL`</span> | Channel to use for fetching RKE2 download URL. Defaults to "latest". Options include: `stable`, `latest`, `testing`. |
```

File: docs/install/install_options/install_options.md [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+| `INSTALL_RKE2_METHOD` | Method of installation to use. Default is on RPM-based systems "rpm", all else "tar". |
```
Comment (cjellick) [inline docs/install/install_options/install_options.md:34]:
```suggestion
| <span class='nowrap'>`INSTALL_RKE2_METHOD`</span> | Method of installation to use. Default is on RPM-based systems "rpm", all else "tar". |
```

File: docs/install/install_options/install_options.md [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+
+
+Environment variables which begin with `RKE2_` will be preserved for the services to use.
```
Comment (cjellick) [inline docs/install/install_options/install_options.md:37]: no they wont

Comment (davidnuzik) [reply docs/install/install_options/install_options.md:37]: thanks. I will remove that line.

Comment (davidnuzik) [reply docs/install/install_options/install_options.md:37]: Or you meant I should warn that they will not be preserved?

**Comment by cjellick** [comment]
going to edit and open a new pr

**Pull request closed without merging**